### PR TITLE
[lxml] Build Fix & Fuzz Targets Improvements

### PR DESCRIPTION
**Pull request by DaveLak**
The context behind this PR is outlined in: https://github.com/google/oss-fuzz/issues/11695

## TL;DR
- Fixes the build.
- Fixes the fuzz targets and improves their efficiency.
- Improves instrumentation of native extensions. 

### Primary fix for the build issues:
- Upgrades the `setuptools` to a minimum version of 42.x (required by `pyinstaller`)
- Upgrades the `pyinstaller` to a minimum of 6.x (v6 introduces helpful improvements in how pyinstaller analyzes package structure and binary dependencies required in the bundled app)
- Both package version ranges are capped at the latest major release to minimize the risk of a new major version breaking the build unexpectedly

### Build Failure Root Cause:
The build check was failing because `pyinstaller` was unable to identify "hidden imports" required by `lxml.etree`
components at runtime. `pyinstaller` requires `setuptools>=42` in order to properly load the `pyinstaller-hooks-contrib` package which includes a collection of hooks specific to `lxml` that allow `pyinstaller` to locate the missing imports.

### Build:
- Use upstream Makefile to configure most of the build config including built dependency versions, hopefully making these tests more valuable and less brittle.
- Fix sanitizer lining of target code and dependencies.
- Build `lxml` dependencies statically, so they are properly instrumented and bundled by Pyinstaller. The bundled fuzz targets now bundles now correctly include hidden imports and build artifacts (e.g., `*.so`, header and C files of built extensions.)
 - The immediate benefit is better instrumentation and feedback provided to the fuzzing engine, but it opens the door for coverage improvements as well.
- Disables the UBSAN shift check (see comment in `build.sh` for context)

#### Other dependency upgrades:
- Use the upstream `lxml/requirements.txt` to dictate the Cpython dependency version (and other dependencies if added.)
- Ensure build tooling is up-to-date.

## Fuzz Harness Improvements:
- Fixes a few typos (e.g., `schema`/`schame`) in variable assignments that were quietly degrading fuzzer efficacy.
- Improve fuzz harness exception handling by
  - Narrowing the error's being swallowed to only those explicitly documented
  - Raising unanticipated exceptions for review 
  - Providing feedback to libfuzzer when an input is not useful
  - Metadata mapping the compiled C code back to the source Cython files are included for improved debugging.
- Clean up the obsolete "`enable_python_coverage=True`" arguments in `atheris.Setup()` calls (this was removed from the Atheris API.)

### Dictionaries & Seed Corpora
- Adds dictionaries and `_seed_corpra.zip`s for existing (in a dedicated repo for now https://github.com/DaveLak/oss-fuzz-inputs) targets to jumpstart the fuzzing engine.
  - Adds custom dictionary for each target based on recommendations produced by the fuzzing engine 
  - Adds a common base dictionary composed of relevant `.dict` files from https://github.com/google/fuzzing. These are merged into a single base dict and appended to each target specific dict.

### CODE REVIEW

**Review by DavidKorczynski — CHANGES_REQUESTED** [review]
- Could you place the dictionaries in a different repository, this is to avoid increasing cloning time of oss-fuzz unnecessarily. 
- Could you avoid changing indentation format?

**Comment by DaveLak** [comment]
> * Could you place the dictionaries in a different repository, this is to avoid increasing cloning time of oss-fuzz unnecessarily.
> * Could you avoid changing indentation format?


Yes and yes. Sorry!

**Comment by DaveLak** [comment]
- I reformatted with YAPF (sorry I missed that, I didn't notice the `.style.yapf` and defaulted to Black.)
- Moved the dictionaries out of this repo and added some reduced seed corpra while I was at it.

Please let me know if I missed anything!

**Comment by DaveLak** [comment]
Sorry about the churn on this; I noticed an issue with the build & bundling that felt worth fixing here.

It should be in good shape now though. I'll leave this branch alone until someone gets a change to review it now.

**Comment by DaveLak** [comment]
I'm going to throw this PR into draft until the ongoing clang upgrade is complete. There seems to be enough involved there without an additional project to check 😅

**Comment by jonathanmetzman** [comment]
David could you please merge this if tests pass?

**Review by DavidKorczynski — APPROVED** [review]

**Comment by DaveLak** [comment]
Thanks!
